javas.js

```diff
@@ -19,4 +20,270 @@ document.querySelectorAll('.nav-link').forEach((n) => n.addEventListener('click'
+  <button class="live"><a href="https://github.com/ijosue16/Mobile-portofolio">See Source </a><span class=><img src="./tools/pop-up/Vector.png" alt=""></span></button>
+</div>
+</div>
+<div class="popup-techs">
+<ul class="mobile-techs">
+<li class="pop-techs">Ruby on rails</li>
+<li class="pop-techs">css</li>
+<li class="pop-techs">JavScript</li>
+</ul>
+
+<ul class="desktop-techs">
+<li class="pop-techs">Codekit</li>
+<li class="pop-techs">GitHub</li>
+<li class="pop-techs">Javasript</li>
+<li class="pop-techs">Bootstrap</li>
+<li class="pop-techs">Terminal</li>
+<li class="pop-techs">Codepen</li>
+</ul>
+</div>
+<p class="popup-para">Lorem Ipsum is simply dummy text of the printing and typesetting industry. 
+Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown 
+printer took a galley of type and scrambled it 1960s with the releaLorem Ipsum is simply dummy 
+text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard 
+dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s with the relea</p>
+
+<div class="popup-mobilebtns">
+<button class="live"><a href="https://ijosue16.github.io/">See Live </a><span class=""><img src="./tools/pop-up/see live icon.svg" alt=""></span></button>
+<button class="live"><a href="https://github.com/ijosue16/Mobile-portofolio">See Source </a><span class=""><img src="./tools/pop-up/Vector.png" alt=""></span></button>
+</div>
+</div>
+`;
+  });
+
+  //
+
+  cardTechnology.append(cardLanguage1, cardLanguage2, cardLanguage3, cardLanguage4);
+  placeHolder.append(cardHeader, cardTechnology, cardButtons);
+  cardTempalte.appendChild(placeHolder);
+  cardContainer.appendChild(cardTempalte);
+  workSection.appendChild(cardContainer);
+}
+
+const emailAddress = document.querySelector('.email');
+const form = document.querySelector('.form');
+const names = document.querySelector('.fullname');
+const message = document.querySelector('.cmt-area');
+const validation = document.createElement('span');
+validation.classList.add('errorMessage');
+
+form.addEventListener('submit', (e) => {
+  form.appendChild(validation);
+  const messages = [];
+  const regex = /^([a-z0-9_-]+)@[a-z0-9-]+(\.[a-z0-9-]+)*(\.[a-z]{2,3})$/;
+  if (names.value.length === 0 || names.value.length === null) {
+    messages.push('⛔ Name is Required');
+    validation.innerHTML = messages.join(',');
+  } else if (names.value.length >= 50) {
+    messages.push('⛔ Name is too long');
+    validation.innerHTML = messages.join(',');
+  } else if (regex.test(emailAddress.value) === false) {
+    messages.push('⛔ Incorrect Email');
+    validation.innerHTML = messages.join(',');
+  } else if (emailAddress.value.length >= 100) {
+    messages.push('⛔ Email Address is too long');
+    validation.innerHTML = messages.join(',');
+  } else if (message.value.length === 0 || message.value.length === null) {
+    messages.push('⛔ Text area is Required');
```

Comment:
- [ OPTIONAL] I think you should be more precise on the errror message as 'The email address should not be in upcase' instead of saying just 'incorrect email' this can confuse users a lil bit, kindly change this error message to be more specific :wink: 
![Screenshot from 2022-11-25 10-31-04](https://user-images.githubusercontent.com/88894525/203936820-d55c0ea2-aa98-4c11-bfe9-1a3bcd91a903.png)